nixos/tests/sunshine.nix

```diff
@@ -0,0 +1,73 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "sunshine";
+  meta = {
+    # test is flaky on aarch64
+    broken = pkgs.stdenv.isAarch64;
+    maintainers = [ lib.maintainers.devusb ];
+  };
+
+  nodes.sunshine = { config, pkgs, ... }: {
+    imports = [
+      ./common/x11.nix
+    ];
+
+    services.sunshine = {
+      enable = true;
+      openFirewall = true;
+      settings = {
+        capture = "x11";
+        encoder = "software";
+        output_name = 0;
+      };
+    };
+
+    environment.systemPackages = with pkgs; [
+      gxmessage
+    ];
+
+  };
+
+  nodes.moonlight = { config, pkgs, ... }: {
+    imports = [
+      ./common/x11.nix
+    ];
+
+    environment.systemPackages = with pkgs; [
+      moonlight-qt
+    ];
+
+  };
+
+  enableOCR = true;
+
+  testScript = ''
+    # start the tests, wait for sunshine to be up
+    start_all()
+    sunshine.wait_for_open_port(48010,"localhost")
+    sunshine.sleep(5)
```

Comment:
Those sleeps could be flaky on high load machines. Can't we rely on the wait_for or retry the commands a couple of times?

Reply:
Not easily -- I didn't really want to use them, but retries are not easily possible since there's GUI involvement for Moonlight, and `wait_for` isn't always reliable because it seems that Sunshine makes ports available before it is _really_ ready -- I wasn't able to find one that reliably indicates that things are ready to go.

Reply:
The test actually seems stable with three of them removed, but the one after Moonlight seems necessary -- since it's a GUI app (and the OCR is _super_ flaky -- it returns different results for the pairing window almost every time) -- I haven't been able to work without a small wait for that one.

Reply:
This might be an opportunity to contribute support for systemd notifications, so we can use `Type=notify` for the service

Reply:
@Scrumplex it's an interesting thought -- not sure it would 100% solve the issue here , as `wait_for_port` seems adequate to understand when Sunshine is ready

Reply:
@SuperSandro2000 I was able to remove all four `sleep`s -- was able to replace the last one with a `wait_for_console_text`. Thanks for the push to make it cleaner.